lib/types.js

```diff
@@ -0,0 +1,70 @@
+/**
```

Comment:
If you prefer, it's possible to write these types-only files as typescript - it's a bit more concise, but also perfectly fine to leave as jsdoc.